tests/test_hpack_echo.c

```diff
@@ -352,3 +348,96 @@ static int test_hpack_stress(struct aws_allocator *allocator, void *ctx) {
 
     return s_tester_clean_up(&s_tester);
 }
+
+AWS_TEST_CASE(localhost_integ_hpack_compression_stress, test_hpack_compression_stress)
+static int test_hpack_compression_stress(struct aws_allocator *allocator, void *ctx) {
+    /* Test that makes tons of streams with all sorts of headers to stress hpack */
```

Comment:
this new test is 95% similar to the other one, right? can we have a common function with a `bool test_compression` arg that toggles the different behavior